cli/src/opts/cast.rs

```diff
@@ -424,37 +424,48 @@ Defaults to decoding output data. To decode input data pass --input or use cast
     },
     #[clap(name = "4byte")]
     #[clap(aliases = &["4", "4b"])]
-    #[clap(about = "Get the function signatures for the given selector from 4byte.directory.")]
+    #[clap(
+        about = "Get the function signatures for the given selector from https://sig.eth.samczsun.com."
+    )]
     FourByte {
         #[clap(help = "The function selector.")]
         selector: String,
     },
     #[clap(name = "4byte-decode")]
     #[clap(aliases = &["4d", "4bd"])]
-    #[clap(about = "Decode ABI-encoded calldata using 4byte.directory.")]
+    #[clap(about = "Decode ABI-encoded calldata using https://sig.eth.samczsun.com.")]
     FourByteDecode {
         #[clap(help = "The ABI-encoded calldata.")]
         calldata: String,
     },
     #[clap(name = "4byte-event")]
     #[clap(aliases = &["4e", "4be"])]
-    #[clap(about = "Get the event signature for a given topic 0 from 4byte.directory.")]
+    #[clap(
+        about = "Get the event signature for a given topic 0 from https://sig.eth.samczsun.com."
+    )]
     FourByteEvent {
         #[clap(help = "Topic 0", value_name = "TOPIC_0")]
         topic: String,
     },
+    #[clap(name = "upload-signature")]
+    #[clap(aliases = &["ups"])]
+    #[clap(about = "Upload the given signatures to https://sig.eth.samczsun.com.")]
+    UploadSignature {
+        #[clap(help = "The signatures to upload. Prefix with 'function', 'event', or 'error'.")]
+        signatures: Vec<String>,
```

Comment:
let's assume you want to batch insert a bunch of function signatures, then it would be required to prefix all with `function`

What would be the most common use case here?

I think it would make sense to also support uploading everything from a contract artifact via cast like `cast ups Greeter.json` perhaps? optionally with an additional filter argument for `functions, event, error, name`?

wdyt @gakonst?

Reply:
> then it would be required to prefix all with function

If you don't give a prefix, it defaults to `function`. I'll add that to the comment

> I think it would make sense to also support uploading everything from a contract artifact

this is a good idea

Reply:
done in https://github.com/foundry-rs/foundry/pull/1716/commits/86a8f80d2f131943fdf63dc9bdcf9d56ed3f36b6